apps/aefate/src/aefa_fate_op.erl

```diff
@@ -1045,3 +1083,13 @@ binary_reverse(Binary) ->
     Size = erlang:size(Binary)*8,
     <<X:Size/integer-little>> = Binary,
     <<X:Size/integer-big>>.
+
+%% TODO: we should check Gas as we go along here...
```

Comment:
I think we should charge gas upfront relative to the size of B. (possibly somewhat related to A also).